lib/src/markdown_processor.dart

```diff
@@ -181,12 +184,73 @@ final List<md.BlockSyntax> _markdownBlockSyntaxes = [
 final RegExp _hideSchemes = RegExp('^(http|https)://');
 
 class MatchingLinkResult {
-  final ModelElement element;
+  final ModelElement modelElement;
   final bool warn;
 
-  MatchingLinkResult(this.element, {this.warn = true});
+  MatchingLinkResult(this.modelElement, {this.warn = true});
+
+  @override
+  bool operator ==(Object other) {
+    return other is MatchingLinkResult &&
+        modelElement == other.modelElement &&
+        warn == other.warn;
+  }
+
+  @override
+  int get hashCode => hash2(modelElement, warn);
+
+  bool isEquivalentTo(MatchingLinkResult other) {
+    if (this == other) return true;
+    if (modelElement?.canonicalModelElement ==
+        other.modelElement?.canonicalModelElement) return true;
+    if (other.modelElement is Parameter && modelElement == null) {
+      return true;
+    }
+    if (modelElement is Parameter && other.modelElement == null) {
+      return true;
+    }
+    return false;
+  }
+
+  @override
+  String toString() {
+    return 'element: ${modelElement?.fullyQualifiedName} warn: $warn';
+  }
 }
 
+class _MarkdownStats {
+  int totalReferences = 0;
+  int resolvedReferences = 0;
+  int resolvedNewLookupReferences = 0;
+  int resolvedOldLookupReferences = 0;
+  int resolvedEquivalentlyReferences = 0;
+
+  String buildReport() {
+    var report = StringBuffer();
+    report.writeln('Reference Counts:');
+    report.writeln('total references: $totalReferences');
+    report.writeln(
+        'resolved references:  $resolvedReferences (${resolvedReferences.toDouble() / totalReferences.toDouble() * 100}%)');
```

Comment:
Might be worth storing `totalReference.toDouble()` as a local, or providing a getter, to shorten these string literals.